edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/edgedeployment/pvc/KubermetesPvcMapper.cs

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Agent.Kubernetes.EdgeDeployment.Pvc
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using k8s.Models;
+    using Microsoft.Azure.Devices.Edge.Agent.Docker.Models;
+    using Microsoft.Azure.Devices.Edge.Util;
+
+    enum PvcOption
+    {
+        None,
+        VolumeName,
+        StorageClass
+    }
+
+    // TODO add unit tests
+    public class KubernetesPvcMapper : IKubernetesPvcMapper
+    {
+        readonly string persistentVolumeName;
+        readonly string storageClassName;
+        readonly uint persistentVolumeClaimSizeMb;
+        readonly PvcOption pvcOption;
+
+        public KubernetesPvcMapper(
+            string persistentVolumeName,
+            string storageClassName,
+            uint persistentVolumeClaimSizeMb)
+        {
+            this.persistentVolumeName = persistentVolumeName;
+            this.storageClassName = storageClassName;
+            this.persistentVolumeClaimSizeMb = persistentVolumeClaimSizeMb;
+            // prefer persistent volume name to storage class name, if both are set.
+            if (!string.IsNullOrWhiteSpace(persistentVolumeName))
+            {
+                this.pvcOption = PvcOption.VolumeName;
+            }
+            else if (!string.IsNullOrWhiteSpace(storageClassName))
+            {
+                this.pvcOption = PvcOption.StorageClass;
+            }
+            else
+            {
+                this.pvcOption = PvcOption.None;
+            }
+        }
+
+        public Option<List<V1PersistentVolumeClaim>> CreatePersistentVolumeClaims(KubernetesModule module, IDictionary<string, string> labels) =>
+            Option.Maybe(module.Config?.CreateOptions?.HostConfig?.Mounts)
+                .Map(mounts => mounts.Select(mount => this.ExtractPvc(mount, labels)).FilterMap().ToList());
+
+        Option<V1PersistentVolumeClaim> ExtractPvc(Mount mount, IDictionary<string, string> labels)
+        {
+            if (!mount.Type.Equals("volume", StringComparison.InvariantCultureIgnoreCase) ||
+                this.pvcOption == PvcOption.None)
+            {
+                return Option.None<V1PersistentVolumeClaim>();
+            }
+
+            string name = KubeUtils.SanitizeK8sValue(mount.Source);
+            bool readOnly = mount.ReadOnly;
+            var persistentVolumeClaimSpec = new V1PersistentVolumeClaimSpec()
+            {
+                AccessModes = new List<string> { readOnly ? "ReadOnlyMany" : "ReadWriteMany" },
+                Resources = new V1ResourceRequirements()
+                {
+                    Requests = new Dictionary<string, ResourceQuantity>() { { "storage", new ResourceQuantity($"{this.persistentVolumeClaimSizeMb}Mi") } }
```

Comment:
new ResourceQuantity($"{this.persistentVolumeClaimSizeMb}**Mi**")
It is **Mi** here but **M** in tests. seems suspicious.

Reply:
`Mi` is 1M as a power of 2, like 1024. `M` is 1M in decimal, i.e. 1000. If I have it inconsistent in the tests, I will fix it.